python/docs/jsonschema_tutorial.rst

```diff
@@ -0,0 +1,161 @@
+containing all point data, and some metadata describing how to decode it:
+
+.. code-block:: json
+
+    "point_stride": { "type": "integer", "minimum": 0, "description": "Number of bytes between points in the `data`" },
+    "fields": {
+      "type": "array",
+      "items": {
+        "$comment": "Generated from PackedElementField by @foxglove/schemas",
+        "title": "PackedElementField",
+        "description": "A field present within each element in a byte array of packed elements.",
+        "type": "object",
+        "properties": {
+          "name": { "type": "string", "description": "Name of the field" },
+          "offset": { "type": "integer", "minimum": 0, "description": "Byte offset from start of data buffer" },
+          "type": {
+            "title": "NumericType: Numeric type",
+            "description": "Type of data in the field. Integers are stored using little-endian byte order.",
+            "oneOf": [
+              { "title": "UNKNOWN", "const": 0 },
+              { "title": "UINT8", "const": 1 },
+              { "title": "INT8", "const": 2 },
+              { "title": "UINT16", "const": 3 },
+              { "title": "INT16", "const": 4 },
+              { "title": "UINT32", "const": 5 },
+              { "title": "INT32", "const": 6 },
+              { "title": "FLOAT32", "const": 7 },
+              { "title": "FLOAT64", "const": 8 }
+            ]
+          }
+        }
+      },
+      "description": "Fields in the `data`"
+    },
+    "data": {
+      "type": "string",
+      "contentEncoding": "base64",
+      "description": "Point data, interpreted using `fields`"
+    }
+
+From the CSV data we have one timestamp and four floating-point data fields per point.
+``foxglove.PointCloud`` uses one timestamp for the whole pointcloud, so we'll just use the first
+point for that. We'll pack each field as a four byte single-precision little-endian float.
+
+We start by describing our data layout in the ``foxglove.PointCloud`` message:
+
+.. literalinclude:: ../examples/jsonschema/pointcloud_csv_to_mcap.py
+    :start-after: # tutorial-point-layout-start
+    :end-before: # tutorial-point-layout-end
+
+And pack the points using the Python built-in ``struct`` and ``base64`` libraries.
+
+.. literalinclude:: ../examples/jsonschema/pointcloud_csv_to_mcap.py
+    :start-after: # tutorial-pack-points-start
+    :end-before: # tutorial-pack-points-end
+
+To set the pointcloud timestamp, the ``foxglove.PointCloud`` schema expects a pair of seconds
+and nanoseconds:
+
+.. literalinclude:: ../examples/jsonschema/pointcloud_csv_to_mcap.py
+    :start-after: # tutorial-timestamp-start
+    :end-before: # tutorial-timestamp-end
+
+And finally we set an arbitrary pose and ``frame_id``.
+
+.. literalinclude:: ../examples/jsonschema/pointcloud_csv_to_mcap.py
+    :start-after: # tutorial-pose-frame-id-start
+    :end-before: # tutorial-pose-frame-id-end
+
+Writing the MCAP
```

Comment:
This post goes back and forth between calling it "the MCAP"/ "an MCAP" vs. "the MCAP file" / "an MCAP file". I think the latter makes more sense.